incubator/hnc/internal/validators/hierarchy.go

```diff
@@ -210,9 +212,59 @@ func (v *Hierarchy) checkParent(ns, curParent, newParent *forest.Namespace) admi
 		return deny(metav1.StatusReasonConflict, "Illegal parent: "+reason)
 	}
 
+	// Prevent overwriting source objects in the descendants after the hierarchy change.
+	if co := v.getConflictingObjects(newParent, ns); len(co) != 0 {
```

Comment:
This seems like a pretty fundamental change in a patch release no? Granted we aren't k8s here, their definition of what is allowed in a patch release: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/release/versioning.md#patch-releases

I totally see why we want this in here, but I'm a bit concerned with setting the precedence.

Reply:
Agreed, this is a large change. Ordinarily I wouldn't want to patch it, but a) it will be a while until v0.6 still and b) I don't think we have enough customers to be _too_ worried yet.

Maybe ask the mailing list and slack channel what they think?

Reply:
Yeah, I think that is a fair compromise.

Reply:
I think we have enough feedback that we can go ahead with this now. But we'll avoid this as HNC matures, and we'll try hard not to get stuck with long-lived branches as well.